notebooks/check_interest.ipynb

```diff
@@ -0,0 +1,201 @@
+    "import numpy as np\n",
+    "\n",
+    "import plotly.figure_factory as ff\n",
+    "import plotly.express as px\n",
+    "import seaborn as sns\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "%load_ext nb_black"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.read_csv(\"../data/input/train.csv\", index_col=\"company_id\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df.info()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def compare_col(df, col1, col2, atol, rtol=0, sample=True):\n",
+    "    mask = np.isclose(df[col1], df[col2], atol=atol, rtol=rtol)\n",
+    "    print(\n",
+    "        f\"Number of rows which have absolute difference more than {atol}: {len(df[~mask])}\"\n",
+    "    )\n",
+    "    if sample:\n",
+    "        return df[~mask].sample(10)\n",
+    "    else:\n",
+    "        return df[~mask]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### calculate interest\n",
+    "\n",
+    "1. use (gross profit + interest)/ total assets\n",
+    "2. use (gross profit + interest)/ sales\n",
+    "\n",
+    "interest_method_1 has only one missing value but only one non-zero value\n",
+    "interest_method_2 has 22 missing values but 5914 non-zero value"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def calculate_interest(df):\n",
+    "    new_df = df.assign(\n",
+    "        **{\n",
+    "            \"total_assets\": 10 ** df[\"logarithm of total assets\"],\n",
+    "            \"sales_method_1\": lambda df: df[\"sales / total assets\"]\n",
+    "            * df[\"total_assets\"],\n",
+    "            \"gross_profit\": lambda df: df[\"gross profit / total assets\"]\n",
+    "            * df[\"total_assets\"],\n",
+    "            \"sales_method_3\": lambda df: df[\"gross_profit\"]\n",
+    "            / df[\"gross profit / sales\"],\n",
+    "            \"sales_method_8\": lambda df: df[\"sales_method_3\"].combine_first(\n",
+    "                df[\"sales_method_1\"]\n",
+    "            ),\n",
+    "            \"interest_method_1\": lambda df: (\n",
+    "                df[\"(gross profit + interest) / total assets\"] * df[\"total_assets\"]\n",
+    "            )\n",
+    "            - df[\"gross_profit\"],\n",
+    "            \"financial_expenses\": lambda df: (\n",
+    "                df[\"(gross profit + interest) / sales\"] * df[\"sales_method_8\"]\n",
+    "            )\n",
+    "            - df[\"gross_profit\"],\n",
+    "        }\n",
+    "    )\n",
+    "    return new_df[[\"interest_method_1\", \"interest_method_2\"]]"
+   ]
```

Comment:
Don't think there is formula for the interest_method_2...?

Reply:
oops the financial expenses should be interest_method_2 sorry!